src/hera/workflows/_runner/script_annotations_util.py

```diff
@@ -87,38 +93,48 @@ def get_annotated_output_param(param_annotation: Parameter) -> Path:
     return path
 
 
+def get_annotated_output_artifact(artifact_annotation: Artifact) -> Path:
+    if artifact_annotation.path:
+        path = Path(artifact_annotation.path)
+    else:
+        path = _get_outputs_path(artifact_annotation)
+    # Automatically create the parent directory (if required)
+    path.parent.mkdir(parents=True, exist_ok=True)
+    return path
+
+
 def get_annotated_artifact_value(artifact_annotation: Artifact) -> Union[Path, Any]:
     """Get the value of the given Artifact annotation.
 
     If the artifact is an output, return the path it will write to.
-    If the artifact is an input, and it has an ArtifactLoader enum as its loader, return the
-    loaded value according to the predefined behaviour (json obj, path or string), otherwise,
-    if the loader is a Callable, it is used directly to load the value.
+    If the artifact is an input, and one of the `loads` or `load` functions are not None,
+    they are used directly to load the value.
+    Otherwise, if it has an ArtifactLoader enum as its loader, return the
+    loaded value according to the predefined behaviour (json obj, path or string).
 
     As Artifacts are always Annotated in function parameters, we don't need to consider
     the `kwargs` or the function parameter name.
     """
     if artifact_annotation.output:
-        if artifact_annotation.path:
-            path = Path(artifact_annotation.path)
-        else:
-            path = _get_outputs_path(artifact_annotation)
-        # Automatically create the parent directory (if required)
-        path.parent.mkdir(parents=True, exist_ok=True)
-        return path
+        return get_annotated_output_artifact(artifact_annotation)
 
     if not artifact_annotation.path:
         # Path is added to the spec automatically when built. As it isn't present in the annotation itself,
         # we need to add it back in for the runner.
         artifact_annotation.path = artifact_annotation._get_default_inputs_path()
 
+    path = Path(artifact_annotation.path)
+    if artifact_annotation.loads is not None and isinstance(artifact_annotation.loads, Callable):  # type: ignore
+        return artifact_annotation.loads(path.read_text())
+
+    if artifact_annotation.loadb is not None and isinstance(artifact_annotation.loadb, Callable):  # type: ignore
+        return artifact_annotation.loadb(path.read_bytes())
```

Comment:
It looks like we only support callables for loads/loadb, and we have no alternate path for these, so I think we can simplify this? It'll raise a TypeError if the user provided a non-Callable, which is what we want.

```suggestion
    if artifact_annotation.loads is not None:
        return artifact_annotation.loads(path.read_text())

    if artifact_annotation.loadb is not None:
        return artifact_annotation.loadb(path.read_bytes())
```